pkg/resources/user.go

```diff
@@ -213,83 +242,119 @@ func CreateUser(d *schema.ResourceData, meta interface{}) error {
+		user, err := client.Users.Describe(ctx, id)
+		if err != nil {
+			if errors.Is(err, sdk.ErrObjectNotExistOrAuthorized) {
+				log.Printf("[DEBUG] user (%s) not found or we are not authorized. Err: %s", d.Id(), err)
+				d.SetId("")
+				return nil
+			}
+			return diag.FromErr(err)
 		}
-		return err
-	}
 
-	if err := setStringProperty(d, "name", user.Name); err != nil {
-		return err
-	}
-	if err := setStringProperty(d, "comment", user.Comment); err != nil {
-		return err
-	}
-	if err := setStringProperty(d, "login_name", user.LoginName); err != nil {
-		return err
-	}
-	if err := setBoolProperty(d, "disabled", user.Disabled); err != nil {
-		return err
-	}
-	if err := setStringProperty(d, "default_role", user.DefaultRole); err != nil {
-		return err
-	}
+		u, err := client.Users.ShowByID(ctx, id)
+		if err != nil {
+			if errors.Is(err, sdk.ErrObjectNotFound) {
+				d.SetId("")
+				return diag.Diagnostics{
+					diag.Diagnostic{
+						Severity: diag.Warning,
+						Summary:  "Failed to query user. Marking the resource as removed.",
+						Detail:   fmt.Sprintf("User: %s, Err: %s", id.FullyQualifiedName(), err),
+					},
+				}
+			}
+			return diag.FromErr(err)
+		}
 
-	var defaultSecondaryRoles []string
-	if user.DefaultSecondaryRoles != nil && len(user.DefaultSecondaryRoles.Value) > 0 {
-		defaultRoles, _ := strings.CutPrefix(user.DefaultSecondaryRoles.Value, "[\"")
-		defaultRoles, _ = strings.CutSuffix(defaultRoles, "\"]")
-		defaultSecondaryRoles = strings.Split(defaultRoles, ",")
-	}
-	if err = d.Set("default_secondary_roles", defaultSecondaryRoles); err != nil {
-		return err
-	}
-	if err := setStringProperty(d, "default_namespace", user.DefaultNamespace); err != nil {
-		return err
-	}
-	if err := setStringProperty(d, "default_warehouse", user.DefaultWarehouse); err != nil {
-		return err
-	}
-	if user.RsaPublicKeyFp != nil {
-		if err = d.Set("has_rsa_public_key", user.RsaPublicKeyFp.Value != ""); err != nil {
-			return err
+		userParameters, err := client.Parameters.ShowParameters(ctx, &sdk.ShowParametersOptions{
+			In: &sdk.ParametersIn{
+				User: id,
+			},
+		})
+		if err != nil {
+			return diag.FromErr(err)
 		}
+
+		if err := setStringProperty(d, "name", user.Name); err != nil {
+			return diag.FromErr(err)
+		}
+		if err := setStringProperty(d, "comment", user.Comment); err != nil {
+			return diag.FromErr(err)
+		}
+		if err := setStringProperty(d, "login_name", user.LoginName); err != nil {
+			return diag.FromErr(err)
+		}
+		if err := setBoolProperty(d, "disabled", user.Disabled); err != nil {
+			return diag.FromErr(err)
+		}
+		if err := setStringProperty(d, "default_role", user.DefaultRole); err != nil {
+			return diag.FromErr(err)
+		}
+
+		var defaultSecondaryRoles []string
+		if user.DefaultSecondaryRoles != nil && len(user.DefaultSecondaryRoles.Value) > 0 {
+			defaultRoles, _ := strings.CutPrefix(user.DefaultSecondaryRoles.Value, "[\"")
```

Comment:
It's not yet the rework of the whole user resource (this implementation was there before).

Changed nevertheless (had to leave the condition to avoid different behavior before the redesign).